portal/src/main/webapp/js/src/study-view/data/StudyViewProxy.js

```diff
@@ -412,6 +413,117 @@ var StudyViewProxy = (function() {
         return deferred.promise();
     }
 
+    //Change the CNA data base on sample ID
+    function convertCNAData(data) {
+        var converted = {};
+
+        if (data) {
+            for (var i = 0, dataL = data.gene.length; i < dataL; i++) {
+                var caseIds = data.caseIds[i];
+
+                for (var j = 0, caseIdsL = caseIds.length; j < caseIdsL; j++) {
+                    if (!converted.hasOwnProperty(caseIds[j])) {
+                        converted[caseIds[j]] = [];
+                    }
+                    converted[caseIds[j]].push({
+                        alter: data.alter[i],
+                        cytoband: data.cytoband[i],
+                        gene: data.gene[i],
+                        gistic: data.gistic[i]
+                    });
+                }
+            }
+        }
+        return converted;
+    }
+
+    function getCNABasedOnSampleIds(sampleIds) {
+        var data = {
+            alter: [],
+            caseIds: [],
+            cytoband: [],
+            gistic: [],
+            gene: []
+        };
+        if (sampleIds instanceof Array) {
```

Comment:
Function documentation has been added in here : https://github.com/zhx828/cbioportal/commit/7155054ce2735af66db3332330f1e79efd729ec6
Unnecessary has been removed in here: https://github.com/zhx828/cbioportal/commit/02a3320e43b29a881da97f10bdb89b01f7079ba0